### ISSUE TITLE
Environment Addons - Importing CFN from Environment

### ISSUE THREAD

**Issue by mcastilho** [aws/copilot-cli#4419]
When trying to create an Aurora at the environment addons level, I need some references to VpcId and PrivateSubnets from the environment, but since the Addons deploys as a nested stack, I believe I am encountering a race-condition when trying to read Output values from environment, since some resources may have not been created yet.

If I deploy a clean environment first, and only then add the addons folder it works fine....

Am I thinking correctly here?

This was never the case for Workload addons since the environment resources were always created already.

**Comment by efekarakus** [comment]
Hi @mcastilho !

> Am I thinking correctly here?

Your thinking is correct 👌, my teammate @Lou1415926 has been thinking hard about this problem 😅 I'm tagging her here so that she can additional information if necessary.


I have different ways of getting around the problem depending on how the environment is created:

- **[best] If you are using a Copilot managed environment**.  That is if you're not importing a VPC in your environment manifest file, then you can get around this problem by creating the following [`addons.parameters.yml` file](https://aws.github.io/copilot-cli/docs/developing/addons/environment/#customizing-the-parameters-section):
   ```yaml
   Parameters:
     VpcId: !Ref VPC
     PrivateSubnets: !Join [ ',', [ !Ref PrivateSubnet1, !Ref PrivateSubnet2 ] ]
   ```
   And don't forget to update the Aurora template with the following parameters:
   ```yaml
   Parameters:
     App:
      Type: String
    Env:
      Type: String
    VpcId:
      Type: String
    PrivateSubnets:
      Type: String
   ```
   This solution will ensure that the addons stack gets created after the VPC and Private Subnets are created and will pass the IDs down into the nested stack.
   

* **If you are importing an existing VPC**. Then today, our best work around is to create a `Mappings` section in your template such that the vpc ID and private subnets are entered again:
    ```yaml
    Mappings:
      Environments:
        test:
            "VpcId": vpc-1234
            "PrivateSubnets": subnet-1111,subnet-2222
        prod:
            "VpcId": vpc-5678
            "PrivateSubnets": subnet-3333,subnet-4444
    ```

Hope this helps!

**Comment by mcastilho** [comment]
@efekarakus @Lou1415926 

So here is the problem that I am encountering. I had already tried what you have suggested since I saw some new env templates going into the main branch recently, and I had tried the `addons.parameters.yml` but no success

Whenever I put more than one parameters in that file, I get a Template error. Let me show you

```
# copilot/environments/addons/addons.parameters.yml
Parameters:
  VpcId: !Ref VPC
  PrivateSubnets: !Join [ ',', [ !Ref PrivateSubnet1, !Ref PrivateSubnet2 ] ]
```

and then when deploying...

```
> copilot env deploy --name production
✔ Proposing infrastructure changes for the sandbox-production environment.
✘ deploy environment production: check if changeset is empty: create change set copilot-584c9296-5ed9-4d91-b871-2ec084caa920 for stack sandbox-production: ValidationError: Template format error: YAML not well-formed. (line 1077, column 7)
        status code: 400, request id: 2ffe7858-872a-454b-9dc8-3a48dee718c7: describe change set copilot-584c9296-5ed9-4d91-b871-2ec084caa920 for stack sandbox-production: ChangeSetNotFound: ChangeSet [copilot-584c9296-5ed9-4d91-b871-2ec084caa920] does not exist
        status code: 404, request id: 1403b91f-a97a-474c-918e-761e29ac1ebc
```

So I started digging at the source code and then I decided to export `copilot env package` and I noticed that the parameters are being merged into the CFN with improper tab indentation causing the Template error.

```
  AddonsStack:
    Metadata:
      'aws:copilot:description': 'A CloudFormation nested stack for your additional AWS resources'
    Type: AWS::CloudFormation::Stack
    Properties:
      Parameters:
        App: !Ref AppName
        Env: !Ref EnvironmentName
        VpcId: !Ref VPC
PrivateSubnets: !Join [',', [!Ref PrivateSubnet1, !Ref PrivateSubnet2]]

      TemplateURL: 

Outputs:
  VpcId:
    Value: !Ref VPC
    Export:

```

Is this a bug?

**Comment by mcastilho** [comment]
#4402 

I just found this PR that is pending release..... I guess I will wait. Thanks.

**Comment by Lou1415926** [comment]
@mcastilho Sorry for the trouble 🙇🏼‍♀️ and thanks for being patient!

**Comment by Lou1415926** [comment]
This should be fixed in [v1.26.0](https://github.com/aws/copilot-cli/releases/tag/v1.26.0) now!
